src/Data/Foreign/Keys.purs

```diff
@@ -16,7 +16,7 @@ foreign import unsafeKeys :: Foreign -> Array String
 -- | Get an array of the properties defined on a foreign value
 keys :: Foreign -> F (Array String)
 keys value
-  | isNull value = Left $ TypeMismatch (NE.singleton "object") "null"
-  | isUndefined value = Left $ TypeMismatch (NE.singleton "object") "undefined"
+  | isNull value = Left $ NE.singleton $ TypeMismatch "object" "null"
```

Comment:
It might be worth extracting `Left <<< NE.singleton <<< TypeMismatch` into a function here.